src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -166,16 +170,85 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
     };
 
     handleFetchError = (error: Object): void => {
-        const { onUpgradeClick } = this.props;
+        const { currentUser, onUpgradeClick, showUpsell, showUpsellWithPicture } = this.props;
         const shouldShowVersionErrorWithUpsell = !!onUpgradeClick && error.status === 403;
-        this.setState({
-            error: shouldShowVersionErrorWithUpsell ? messages.versionNotAvailable : messages.versionFetchError,
-            errorTitle: shouldShowVersionErrorWithUpsell ? messages.versionAccessError : messages.versionServerError,
-            isLoading: false,
-            isWatermarked: false,
-            versionCount: 0,
-            versions: [],
-        });
+        const shouldShowVersions = showUpsell || showUpsellWithPicture;
```

Comment:
posted a similar comment in GHE, but we're passing in multiple different upsell props and i don't think it's obvious to the caller which take precedent (e.g. what happens if all are true?)

is there a way to consolidate all potential upsell variations into a single prop?